From 9ac417b2a5ad6291046c16c9219a0b284f1699bf Mon Sep 17 00:00:00 2001
From: Loic Poulain <loic.poulain@intel.com>
Date: Mon, 8 Sep 2014 14:26:55 +0200
Subject: [PATCH] Revert "ACPI / LPSS: fix UART Auto Flow Control"

Issue: IRDA-935 LCK-1300

This reverts commit 088f1fd267c7f43b5d87850a0fa0c7e851ecae97.

This reverted patch introduces a RX perf regression.
We receive too much irq causing many small DMA transfers (16B)
(character timeout irq).

It is safe to revert this patch because the "chicken bit" that is
the root cause of the problem, should only be used when auto-flow
control is not used, and we always use auto-flow control.

Signed-off-by: Loic Poulain <loic.poulain@intel.com>
---
 drivers/acpi/acpi_lpss.c |   12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

--- linux.orig/drivers/acpi/acpi_lpss.c
+++ linux/drivers/acpi/acpi_lpss.c
@@ -31,7 +31,6 @@ ACPI_MODULE_NAME("acpi_lpss");
 /* Offsets relative to LPSS_PRIVATE_OFFSET */
 #define LPSS_GENERAL			0x08
 #define LPSS_GENERAL_LTR_MODE_SW	BIT(2)
-#define LPSS_GENERAL_UART_RTS_OVRD	BIT(3)
 #define LPSS_SW_LTR			0x10
 #define LPSS_AUTO_LTR			0x14
 #define LPSS_TX_INT			0x20
@@ -77,16 +76,11 @@ struct lpss_private_data {
 
 static void lpss_uart_setup(struct lpss_private_data *pdata)
 {
-	unsigned int offset;
+	unsigned int tx_int_offset = pdata->dev_desc->prv_offset + LPSS_TX_INT;
 	u32 reg;
 
-	offset = pdata->dev_desc->prv_offset + LPSS_TX_INT;
-	reg = readl(pdata->mmio_base + offset);
-	writel(reg | LPSS_TX_INT_MASK, pdata->mmio_base + offset);
-
-	offset = pdata->dev_desc->prv_offset + LPSS_GENERAL;
-	reg = readl(pdata->mmio_base + offset);
-	writel(reg | LPSS_GENERAL_UART_RTS_OVRD, pdata->mmio_base + offset);
+	reg = readl(pdata->mmio_base + tx_int_offset);
+	writel(reg | LPSS_TX_INT_MASK, pdata->mmio_base + tx_int_offset);
 }
 
 static struct lpss_device_desc lpt_dev_desc = {
